### PR TITLE
gRPC Server: Instrument requests made to the server.

### PR DESCRIPTION
**Pull request by stevesg**
Expose metrics from the gRPC server in order to monitor for failed responses
and response latency. Uses code from the already vendored weaveworks/common.

### CODE REVIEW

**Review by JohnnyQQQQ — APPROVED** [review]
LGTM

File: pkg/services/grpcserver/service.go
Comment (sakjur) [inline pkg/services/grpcserver/service.go:1]: On line 42, you can get a registerer (currently that's Prometheus's default registerer, so same as `prometheus.Register`), which should make it easier for you to test your metrics through unit tests.

```
func ProvideService(cfg *setting.Cfg, authenticator interceptors.Authenticator, tracer tracing.Tracer, registerer prometheus.Registerer) (Provider, error) {
```

**Review by mildwonkey — APPROVED** [review]
👍 I'm not sure if I can easily test this out myself, but the code looks reasonable to me

**Review by sakjur — APPROVED** [review]
+1, this seems like a good thing to do and the code looks good, I've added a couple of comments about the registration of the metrics. I have not tested running the changes, fwiw.

**Pull request merged**